src/components/spreadsheet/spreadsheet-gs-filter.tsx

```diff
@@ -0,0 +1,103 @@
+}
+
+export const SpreadsheetGsFilter = ({ equipmentType, applyGsFilter }: SpreadsheetGsFilterProps) => {
+    const dispatch = useDispatch();
+    const gsFilterSpreadsheetState = useSelector((state: AppState) => state.gsFilterSpreadsheetState);
+    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+    const filterConfigOpen = useStateBoolean(false);
+
+    const formMethods = useForm<SpreadsheetGsFilterForm>({
+        defaultValues: initialSpreadsheetGsFilterForm,
+        resolver: yupResolver(spreadsheetGsFilterFormSchema),
+    });
+    const { reset, watch } = formMethods;
+    const spreadsheetGsFilterWatcher = watch(SPREADSHEET_GS_FILTER);
+
+    const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+        setAnchorEl(event.currentTarget);
+        filterConfigOpen.setTrue();
+    };
+
+    const handleClose = useCallback(() => {
+        setAnchorEl(null);
+        filterConfigOpen.setFalse();
+        dispatch(saveSpreadsheetGsFilters(equipmentType, toExpertFilter(spreadsheetGsFilterWatcher)));
+    }, [dispatch, equipmentType, filterConfigOpen, spreadsheetGsFilterWatcher]);
+
+    useEffect(() => {
+        const filters = gsFilterSpreadsheetState[equipmentType] || [];
+        reset(toFormFormat(filters));
+        applyGsFilter(filters);
+    }, [equipmentType, reset, gsFilterSpreadsheetState, applyGsFilter]);
+
+    return (
+        <>
+            <Badge
+                color="secondary"
+                badgeContent={spreadsheetGsFilterWatcher?.length}
+                anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
+            >
+                <Button
+                    sx={spreadsheetStyles.spreadsheetButton}
+                    size="small"
+                    onClick={handlePopoverOpen}
+                    startIcon={<ArticleIcon />}
+                >
+                    <FormattedMessage id="spreadsheet/filter/config" />
+                </Button>
+            </Badge>
+
+            <Popover
+                open={filterConfigOpen.value}
+                anchorEl={anchorEl}
+                onClose={handleClose}
+                anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
+                transformOrigin={{ vertical: 'top', horizontal: 'center' }}
+                slotProps={{ paper: { sx: { minWidth: '200px' } } }}
+            >
+                <CustomFormProvider validationSchema={spreadsheetGsFilterFormSchema} {...formMethods}>
+                    <DirectoryItemsInput
+                        name={SPREADSHEET_GS_FILTER}
+                        titleId="FiltersListsSelection"
+                        label="filter"
+                        elementType={ElementType.FILTER}
+                        equipmentTypes={[equipmentType]}
+                        labelRequiredFromContext={false}
+                    />
+                </CustomFormProvider>
+            </Popover>
+        </>
+    );
```

Comment:
Maybe we should only have the DirectoryItemsInput no ? I think it's a bit weird to have this multiselect input opened after a click on a button

Reply:
I thought it would be cleaner and more homogeneous to add a button instead of a field directly in the spreadsheet top bar but I agree a popover only containing one field might be out of place in the application. I could check with Christine

Reply:
Yeah maybe you can see that with Christine :)

Reply:
She agreed with you I'll roll this back